pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -257,45 +290,46 @@ func createDnsmasqConfiglet(
 		if advertizeRouter && !ctx.disableDHCPAllOnesNetMask {
 			// Network prefix "255.255.255.255" will force packets to go through
 			// dom0 virtual router that makes the packets pass through ACLs and flow log.
-			file.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
+			io.WriteString(buffer, fmt.Sprintf("dhcp-option=option:netmask,%s\n",
 				"255.255.255.255"))
 		} else {
-			file.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
+			io.WriteString(buffer, fmt.Sprintf("dhcp-option=option:netmask,%s\n",
 				ipv4Netmask))
 		}
 	}
 	if advertizeRouter {
 		// IPv6 XXX needs to be handled in radvd
 		if !isIPv6 {
-			file.WriteString(fmt.Sprintf("dhcp-option=option:router,%s\n",
+			io.WriteString(buffer, fmt.Sprintf("dhcp-option=option:router,%s\n",
 				router))
 			if !ctx.disableDHCPAllOnesNetMask {
-				file.WriteString(fmt.Sprintf("dhcp-option=option:classless-static-route,%s/32,%s,%s,%s,%s,%s\n",
+				io.WriteString(buffer, fmt.Sprintf("dhcp-option=option:classless-static-route,%s/32,%s,%s,%s,%s,%s\n",
 					router, "0.0.0.0",
 					"0.0.0.0/0", router,
 					netstatus.Subnet.String(), router))
 			}
 		}
 	} else {
-		log.Functionf("createDnsmasqConfiglet: no router\n")
+		log.Functionf("createDnsmasqConfigletWithBuffer: no router\n")
```

Comment:
Some logs refer to this function as `createDnsmasqConfigletWithBuffer` (instead of `createDnsmasqConfigletToWriter`)